Add section about draft PRs to handbook (#431)
sourcegraph/about: handbook/engineering/code_reviews.md

GitHub will automatically assign reviewers if there is a matching entry in the [CODEOWNERS](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/.github/CODEOWNERS) file, but that doesn't necessarily mean that you need to wait for an approval from everyone. For example, if you are making a change to the search backend then you only need approval from one person on that team, not all of them.

Use your best judgement and ask if you are uncertain.

## When should I use a draft PR?

> A draft PR signals that the change is not ready for reviewed. If you are looking for feedback or discussion on your change, you should mark the PR as ready for review, even if you expect more work to be done on the change.

Reviewer: Suggest making this change and then deleting the bullet below.

```suggestion
A draft PR signals that the change is not ready for reviewed. This is useful, for example, if you want to self-review your diff before sending review requests to others. If you are looking for feedback or discussion on your change, then you should mark the PR as ready for review and communicate your intentions in the PR description.
```

> Draft PRs are great for:

Author: ```suggestion
```

> 

Author: ```suggestion
```

> - Doing a self-review (see [effective code reviews](#what-makes-an-effective-code-review) above).

Author: ```suggestion
```
